src/dcc.c

```diff
@@ -2239,27 +2239,40 @@ struct dcc_table DCC_IDENTWAIT = {
 
 void dcc_ident(int idx, char *buf, int len)
 {
-  char response[512], uid[512], buf1[UHOSTLEN + 21];
+  unsigned int port_on_server = 0, port_on_client = 0;
+  char resp_type[512], add_info[512], user_id[512], *s;
+  size_t l;
   int i;
 
-  *response = *uid = '\0';
-  sscanf(buf, "%*[^:]:%[^:]:%*[^:]:%[^\n]\n", response, uid);
-  rmspace(response);
-  if (response[0] != 'U') {
-    dcc[idx].timeval = now;
+  *resp_type = *add_info = *user_id = '\0'; 
+  if ((sscanf(buf, "%u ,%u : %511[^ :] : %511[^ :] :%511s", &port_on_server,
```

Comment:
```
Notes on Syntax:

   1)   To promote interoperability among variant
        implementations, with respect to white space the above
        syntax is understood to embody the "be conservative in
        what you send and be liberal in what you accept"
        philosophy.  Clients and servers should not generate
        unnecessary white space (space and tab characters) but
        should accept white space anywhere except within a
        token.  In parsing responses, white space may occur
        anywhere, except within a token.
```
Coming from an IRC background it is very important to follow the "be liberal in what you accept" rule with so many ircds out there, so we should also do this for ident, which means not requiring those spaces to be there

Reply:
exactly. i did read the same paragraph and implemented it. thats why i put those whitespaces into the sscanf(). those whitespaces in sscanf() do not mean "exactly one whitespace". instead they mean "any amount of whitespace" including "no whitespace". i developer tested with strings containing zero whitespace and strings containing 3 consecutive whitespaces.